packages/devtools_app/lib/src/memory/memory_heap_tree_view.dart

```diff
@@ -469,48 +489,170 @@ class HeapTreeViewState extends State<HeapTree>
+          child: OutlineButton(
+            key: allocationMonitorResetKey,
+            onPressed: () async {
+              await _allocationReset();
+            },
+            child: createImageIcon(
+              'icons/memory/reset_icon_white@2x.png',
+              size: defaultIconThemeSize,
+            ),
+          ),
+        ),
       ],
     );
   }
 
+  Future<void> _allocationStart() async {
+    // TODO(terry): Look at grouping by library or classes also filtering e.g., await controller.computeLibraries();
+    controller.memoryTimeline.addMonitorStartEvent();
+
+    final allocationtimestamp = DateTime.now();
+    final currentAllocations = await controller.getAllocationProfile();
+
+    if (controller.monitorAllocations.isNotEmpty) {
+      final previousLength = controller.monitorAllocations.length;
+      int previousIndex = 0;
+      final currentLength = currentAllocations.length;
+      int currentIndex = 0;
+      while (currentIndex < currentLength && previousIndex < previousLength) {
+        final previousAllocation = controller.monitorAllocations[previousIndex];
+        final currentAllocation = currentAllocations[currentIndex];
+
+        if (previousAllocation.classRef.id == currentAllocation.classRef.id) {
+          final instancesCurrent = currentAllocation.instancesCurrent;
+          final bytesCurrent = currentAllocation.bytesCurrent;
+
+          currentAllocation.instancesAccumulated =
+              previousAllocation.instancesAccumulated +
+                  (instancesCurrent - previousAllocation.instancesCurrent);
+          currentAllocation.bytesAccumulated =
+              previousAllocation.bytesAccumulated +
+                  (bytesCurrent - previousAllocation.bytesCurrent);
+
+          final instancesAccumulated = currentAllocation.instancesAccumulated;
+          final bytesAccumulated = currentAllocation.bytesAccumulated;
+
+          if (instancesAccumulated != 0 || bytesAccumulated != 0) {
+            debugLogger('previous,index=[$previousIndex][$currentIndex] '
+                'class ${currentAllocation.classRef.name}\n'
+                '    instancesCurrent=$instancesCurrent, '
+                '    instancesAccumulated=${currentAllocation.instancesAccumulated}\n'
+                '    bytesCurrent=$bytesCurrent, '
+                '    bytesAccumulated=${currentAllocation.bytesAccumulated}}\n');
+          }
+
+          previousIndex++;
+          currentIndex++;
+        } else {
+          // Either a new class in currentAllocations or old that's no longer
+          // active in previousAllocations.
+          final currentClassId = currentAllocation.classRef.id;
+          final ClassHeapDetailStats first =
+              controller.monitorAllocations.firstWhere(
+            (element) => element.classRef.id == currentClassId,
+            orElse: () => null,
+          );
+          if (first != null) {
+            //It's a class that's gone?
```

Comment:
missing space and replace `?` with `.` as hopefully that is a statement about what is happening not a question.

Reply:
// A class that no longer exist (live or sentinel).